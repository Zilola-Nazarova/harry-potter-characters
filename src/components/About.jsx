import styles from '@/styles/About.module.css'

const About = () => {
  return (
    <div className={styles.about}>About this Website</div>
  )
}

export default About;
