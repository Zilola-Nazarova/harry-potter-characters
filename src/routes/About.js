import styles from '../styles/About.module.css';

const About = () => (
  <div className={styles.about}>
    <h2 className={styles.bright}>What is Mountains Forecast?</h2>
    <p>
      Mountains Forecast is a site that lists down the mountains
      located in Uzbekistan with the following data for each
      mountain:
    </p>
    <ul className={styles.dark}>
      <li>elevation (in m)</li>
      <li>weather code (index)</li>
    </ul>
    <p>
      When a mountain is selected, current weather
      and hourly forecasts are displayed:
    </p>
    <ul>
      <li className={styles.bright}>
        <span>current weather:</span>
        <ul>
          <li>temperature (in °C)</li>
          <li>wind speed (in km/h)</li>
          <li>wind direction (in °)</li>
        </ul>
      </li>
      <li className={styles.bright}>
        <span>hourly forecast:</span>
        <ul>
          <li>temperature (in °C)</li>
          <li>precipitation (in mm)</li>
          <li>wind speed (in km/h)</li>
          <li>cloudcover (in %)</li>
        </ul>
      </li>
    </ul>
    <p>
      You can swith the displayed forecast to show information
      according to ICON or GFS forecast models.
    </p>
    <p>
      Every page containing the weather data displays
      information about the last update under the navigation
      bar.
    </p>
    <p>
      The navigation bar has Models button with links to
      information about main forecast models: ICON and GFS.
    </p>

    <div className={styles.credits}>
      <h3 className={styles.bright}>thanks</h3>
      <p>
        Weather information by
        {' '}
        <a target="_blank" rel="noreferrer" href="https://open-meteo.com/">
          open-meteo.com
        </a>
        API.
      </p>
      <p>
        Map of Uzbekistan by
        {' '}
        <a target="_blank" rel="noreferrer" href="https://vemaps.com/">
          Vemaps.com
        </a>
        .
      </p>
      <p>
        Original design idea by
        {' '}
        <a target="_blank" rel="noreferrer" href="https://www.behance.net/sakwadesignstudio">
          Nelson Sakwa
        </a>
        on Behance.
      </p>
    </div>
  </div>
);

export default About;
