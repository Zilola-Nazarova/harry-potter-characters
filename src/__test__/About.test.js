import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import About from '../routes/About';

describe('Test About render', () => {
  test('Test About', () => {
    const component = renderer.create(
      <BrowserRouter>
        <About />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
