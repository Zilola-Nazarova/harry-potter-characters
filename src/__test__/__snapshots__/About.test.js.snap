// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test About render Test About 1`] = `
<div
  className="about"
>
  <h2
    className="bright"
  >
    What is Harry Potter Lybrary?
  </h2>
  <p>
    Harry Potter Lybrary is a site that lists down main Harry Potter characters with the following data:
  </p>
  <ul
    className="dark"
  >
    <li>
      Name
    </li>
    <li>
      House
    </li>
    <li>
      Alive/Not alive
    </li>
    <li>
      Character's portrait
    </li>
  </ul>
  <p>
    When a character is selected, more data about this character is displayed such as:
  </p>
  <ul>
    <li>
      Wand properties
    </li>
    <li>
      Patronus animal
    </li>
    <li>
      Date of birth
    </li>
    <li>
      Ancestry
    </li>
  </ul>
  <p>
    The home page has a search bar where you can type your character's name to find the information faster.
  </p>
  <div
    className="credits"
  >
    <h3
      className="bright"
    >
      thanks
    </h3>
    <p>
      Harry Potter characters information by
       
      <a
        href="https://hp-api.onrender.com/"
        rel="noreferrer"
        target="_blank"
      >
        hp-api.onrender.com/
      </a>
      API.
    </p>
    <p>
      Original design idea by
       
      <a
        href="https://www.behance.net/sakwadesignstudio"
        rel="noreferrer"
        target="_blank"
      >
        Nelson Sakwa
      </a>
      on Behance.
    </p>
  </div>
</div>
`;
